CactbotOverlay/FFXIVProcess.cs

```diff
@@ -703,34 +729,125 @@ public enum Song {
         return o != null &&
           song_ms == o.song_ms &&
           song_procs == o.song_procs &&
+          soul_gauge == o.soul_gauge &&
           song_type == o.song_type;
       }
 
       public override int GetHashCode() {
         int hash = 17;
         hash = hash * 31 + song_ms.GetHashCode();
         hash = hash * 31 + song_procs.GetHashCode();
+        hash = hash * 31 + soul_gauge.GetHashCode();
         hash = hash * 31 + song_type.GetHashCode();
         return hash;
       }
     }
 
-    public BardJobData GetBard() {
-      byte[] bytes = GetJobSpecificData();
-      if (bytes == null)
-        return null;
+    public unsafe BardJobData GetBard() {
+      fixed (byte* p = GetJobSpecificData()) {
+        if (p == null)
+          return null;
+        else {
+          BardJobMemory mem = *(BardJobMemory*)&p[0];
+          BardJobData j = new BardJobData() {
+            song_ms = mem.song_ms,
+            song_procs = mem.song_procs,
+            soul_gauge = mem.soul_gauge,
+            song_type = (BardJobData.Song)mem.song_type,
+          };
+          return j;
+        }
+      }
+    }
+
+    [StructLayout(LayoutKind.Explicit)]
+    public unsafe struct DancerJobMemory { // TODO: Needs more research. 
```

Comment:
As with gunbreaker, I'm not 100% confident with these, as there could be more going on than I was able to see at level 60.